### PR TITLE
📝 [RUM-158] Add jsdoc annotations on initConfiguration

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

JSDoc annotations are visible through the IDE, providing developers with immediate and comprehensive documentation as they integrate with our SDK. 

Related to https://github.com/DataDog/browser-sdk/issues/1531

## Changes

Add jsdoc comments on initConfiguration object

## Testing

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by cy-moi — APPROVED** [review]

File: packages/core/src/domain/configuration/configuration.ts [hunk 1]
```diff
@@ -30,43 +30,126 @@ export const TraceContextInjection = {
...
+   */
   silentMultipleInit?: boolean | undefined
+  /**
+   * Allows the use of localStorage when cookies cannot be set. This enables the RUM Browser SDK to run in environments that do not provide cookie support.
+   * See [Monitor Electron Applications Using the Browser SDK](https://docs.datadoghq.com/real_user_monitoring/guide/monitor-electron-applications-using-browser-sdk) for further information.
+   * @default false
+   */
   allowFallbackToLocalStorage?: boolean | undefined
+  /**
+   * Allow untrusted events to be collected by the browser SDK. Can be useful if you heavily rely on programmatic events, such as in an automated UI test environment.
```
Comment (BenoitZugmeyer) [inline packages/core/src/domain/configuration/configuration.ts:60]: **💬 suggestion:** ‏
```suggestion
   * Allow listening to DOM events dispatched programmatically (untrusted events). Enabling this option can be useful if you heavily rely on programmatic events, such as in an automated UI test environment.
```

I would also add a link to https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted

File: packages/core/src/domain/configuration/configuration.ts [hunk 2]
```diff
@@ -30,43 +30,126 @@ export const TraceContextInjection = {
...
+  /**
+   * [Internal option] The percentage of telemetry configuration sent. A value between 0 and 100.
+   * @default 5
+   */
   telemetryConfigurationSampleRate?: number
+  /**
+   * [Internal option] The percentage of telemetry usage sent. A value between 0 and 100.
+   * @default 5
+   */
   telemetryUsageSampleRate?: number
```
Comment (BenoitZugmeyer) [inline packages/core/src/domain/configuration/configuration.ts:153]: **💭 thought:** ‏I'm a bit surprised to see those options are internal. I'm fine with it, I just thought those were user-facing.

File: packages/logs/src/domain/configuration.ts [hunk 3]
```diff
@@ -15,10 +15,31 @@ import type { LogsEvent } from '../logsEvent.types'
...
+   * Forward console.error logs, uncaught exceptions and network errors to Datadog.
+   * @default true
+   */
   forwardErrorsToLogs?: boolean | undefined
+  /**
+   * Forward logs from console.* to Datadog. Use "all" to forward everything or an array of console API names to forward only a subset.
+   */
   forwardConsoleLogs?: ConsoleApiName[] | 'all' | undefined
+  /**
+   * Forward reports from the Reporting API to Datadog. Use "all" to forward everything or an array of report types to forward only a subset.
```
Comment (BenoitZugmeyer) [inline packages/logs/src/domain/configuration.ts:36]: **💬 suggestion:** ‏a link to https://developer.mozilla.org/en-US/docs/Web/API/Reporting_API could be nice

File: packages/core/src/domain/configuration/configuration.ts (hunk 2 above)
Comment (amortemousque) [reply packages/core/src/domain/configuration/configuration.ts:153]: IIRC, the discussion I had with Bastien when introducing `telemetryConfigurationSampleRate`, the idea was that we didn't want to overcomplicate the telemetry on the customer side and only advocate for `telemetrySampleRate`.

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**